packages/advisor-components/src/ContextWrapper.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { IntlProvider } from 'react-intl';
+import { Router } from 'react-router-dom';
+import { History } from 'history';
+
+import { AdvisorProduct } from './types';
+
+export const translations: Record<string, Record<string, string>> = {
+  en: require('../../compiled-lang/en.json'),
+};
+
+export interface ContextWrapperProps {
+  /**
+   * routerProps - passed to Router from react-router-dom
+   */
+  routerProps?: {
+    history: History;
+  };
+  env: AdvisorProduct;
+  children: React.ReactNode;
+}
+
+export const ProductContext = React.createContext(AdvisorProduct.rhel);
+
+const ContextWrapper: React.FC<ContextWrapperProps> = ({ routerProps, env, children }) => {
+  const locale = navigator.language.slice(0, 2);
+
+  return routerProps ? (
+    <Router {...routerProps}>
+      <IntlProvider locale={locale} messages={translations[locale]}>
+        <ProductContext.Provider value={env}>{children}</ProductContext.Provider>
+      </IntlProvider>
+    </Router>
+  ) : (
+    <IntlProvider locale={locale} messages={translations[locale]}>
+      <ProductContext.Provider value={env}>{children}</ProductContext.Provider>
```

Comment:
Wouldn't it be better to pass the `env` values directly to the components? I see the usage just in `RuleDetailsBase` and we can pass the `env` directly to it.

Reply:
OK, I think it's simpler 👍🏼